### ISSUE TITLE
[AvalonDock] Crash when dragging and dropping repeatedly with nested DockingManagers

### ISSUE THREAD

**Issue by nssigma** [xceedsoftware/wpftoolkit#1345]
I am using  Extended.Wpf.Toolkit.3.4.0.
Application errors occur when Docking or displaying a floating window.
UserControl with DockingManager is displayed inside DockingManager as contents of LayoutAnchorable.

This case is different from [#1142](https://github.com/xceedsoftware/wpftoolkit/issues/1142).

There is the same bug in Ver.3.7.18313.20590.

- MainWindow.xaml

```
<Window ...>
 <Grid>
  <xcad:DockingManager >
   <xcad:LayoutRoot>
    <xcad:LayoutPanel>
     <xcad:LayoutDocumentPane>
      <xcad:LayoutAnchorable Title="Usercontrol1">
       <local:UserControl1 />
      </xcad:LayoutAnchorable>
      <xcad:LayoutAnchorable Title="Usercontrol2">
       <local:UserControl1  />
      </xcad:LayoutAnchorable>
      <xcad:LayoutAnchorable Title="Usercontrol3">
       <local:UserControl1  />
      </xcad:LayoutAnchorable>
     </xcad:LayoutDocumentPane>
    </xcad:LayoutPanel>
   </xcad:LayoutRoot>
  </xcad:DockingManager>
 </Grid>
</Window>
```

- UserControl1.xaml

```
<UserControl ...>
 <Grid>
  <xcad:DockingManager >
   <xcad:LayoutRoot>
    <xcad:LayoutPanel>
     <xcad:LayoutDocumentPane>
      <xcad:LayoutAnchorable Title="Text1">
       <TextBlock Text="text1"/>
      </xcad:LayoutAnchorable>
      <xcad:LayoutAnchorable Title="Text2">
       <TextBlock Text="text2"/>
      </xcad:LayoutAnchorable>
      <xcad:LayoutAnchorable Title="Text3">
       <TextBlock Text="text3"/>
      </xcad:LayoutAnchorable>
     </xcad:LayoutDocumentPane>
    </xcad:LayoutPanel>
   </xcad:LayoutRoot> 
  </xcad:DockingManager>
 </Grid>
</UserControl>
```

Stacktrace:
説明: ハンドルされない例外のため、プロセスが中止されました。
例外情報:System.ArgumentOutOfRangeException
   場所 System.ThrowHelper.ThrowArgumentOutOfRangeException(System.ExceptionArgument, System.ExceptionResource)
   場所 System.Collections.ObjectModel.Collection`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Insert(Int32, System.__Canon)
   場所 Xceed.Wpf.AvalonDock.Controls.DocumentPaneDropTarget.Drop(Xceed.Wpf.AvalonDock.Layout.LayoutAnchorableFloatingWindow)
   場所 Xceed.Wpf.AvalonDock.Controls.DropTarget`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].Drop(Xceed.Wpf.AvalonDock.Layout.LayoutFloatingWindow)
   場所 Xceed.Wpf.AvalonDock.Controls.DragService.Drop(System.Windows.Point, Boolean ByRef)
   場所 Xceed.Wpf.AvalonDock.Controls.LayoutFloatingWindowControl.FilterMessage(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   場所 Xceed.Wpf.AvalonDock.Controls.LayoutAnchorableFloatingWindowControl.FilterMessage(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   場所 System.Windows.Interop.HwndSource.PublicHooksFilterMessage(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   場所 MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   場所 MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   場所 System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   場所 System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   場所 System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   場所 MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)

Sample Source
[WpfAvalonDockTest4_source.zip](https://github.com/xceedsoftware/wpftoolkit/files/2183061/WpfAvalonDockTest4_source.zip)

Capture Movie
[AvalonDockTest_movie.zip](https://github.com/xceedsoftware/wpftoolkit/files/2183062/AvalonDockTest_movie.zip)

**Comment by XceedBoucherS** [comment]
This will be fixed in v3.8.

**Comment by nssigma** [comment]
Thanks. 
I will wait for the release of v3.8.
